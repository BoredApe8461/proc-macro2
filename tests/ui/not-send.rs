use proc_macro2::Span;

fn main() {
    fn requires_send<T: Send>() {}
    requires_send::<Span>();
}
