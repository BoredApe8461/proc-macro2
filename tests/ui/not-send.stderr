error[E0277]: `proc_macro::Span` cannot be sent between threads safely
 --> $DIR/not-send.rs:5:5
  |
4 |     fn requires_send<T: Send>() {}
  |                         ---- required by this bound in `requires_send`
5 |     requires_send::<Span>();
  |     ^^^^^^^^^^^^^^^^^^^^^ `proc_macro::Span` cannot be sent between threads safely
  |
  = help: within `Span`, the trait `Send` is not implemented for `proc_macro::Span`
  = note: required because it appears within the type `proc_macro2::imp::Span`
  = note: required because it appears within the type `Span`

error[E0277]: `Rc<()>` cannot be sent between threads safely
 --> $DIR/not-send.rs:5:5
  |
4 |     fn requires_send<T: Send>() {}
  |                         ---- required by this bound in `requires_send`
5 |     requires_send::<Span>();
  |     ^^^^^^^^^^^^^^^^^^^^^ `Rc<()>` cannot be sent between threads safely
  |
  = help: within `Span`, the trait `Send` is not implemented for `Rc<()>`
  = note: required because it appears within the type `PhantomData<Rc<()>>`
  = note: required because it appears within the type `Span`
